### PR TITLE
Makes Fire Hot Again

### PR DESCRIPTION
**Pull request by Neerti**
Being lit on fire barely did anything, for multiple reasons, all which should be fixed in this PR.
First, the code to do damage based on bodytemperature was literally bugged and only applied the lowest level of damage, no matter what.
Second, wounds made autohealed them fairly fast, sometimes even outhealing the incoming damage.
Third, being on fire raised your temperature rather slowly.
Fourth, the lowest level of 'overheating' damage was rather low.

Changes:
The bodytemperature part of Life() is now unbroken and I accidentally fixed freezing not hurting, so yay.
Added a ten minute delay to wounds being able to autoheal them away.  Applying bandage/salves will skip this.
Temperature climb is now based on fire stacks while on fire, so more stacks means more and longer pain.
Lowest level of overheating was tweaked.
Running into people while on fire will split your fire stacks with them and light them on fire.
Ports /tg/'s on fire sprite.
Different mobs can have different icons for being on fire.
Bonus: Rejuv now removes pain.

### CODE REVIEW

**Comment by Dankman** [comment]
how does your change affect chemicals that affect body temperature? prediction: you just also accidentally made temperature chemicals obscenely powerful murder machines

**Comment by Wickedtemp** [comment]
Don't worry about temperature changing chems. All Medical/Science really has in that department is frost oil (and potentially condensed capsaicin? I'm unsure, I'll have to check), which they can't get without a botanist, and leporazine, which stabilizes temperature and nothing else. 

And then fire. But if you're on fire, you have bigger problems to worry about than just raised temperature :p

**Comment by Wickedtemp** [comment]
Question: Does autoheal begin ten minutes after the wound was made, or is it ten minutes after any damage is taken? If they get hit again, does that reset the timer?

**Comment by Dankman** [comment]
Does this make enchiladas deadly?

**Comment by Neerti** [comment]
If I remember correctly chemicals like frost oil and cryotoxin have caps to not go below a specific temperature too.

**Comment by Dankman** [comment]
before merging, could you check? Cause like, changing how the body handles temperature, changes a LOT of stuff, Polaris has fullbody prosthetics, right? Being out in vacuum without a cooler is usually a pretty hard death sentence, but with these buffs to spike damage, it could get really bad, really fast. 

This is a change to a core mechanic that a lot of the game was built around, while leaving everything that was designed with the original functionality in place untouched.

**Comment by Wickedtemp** [comment]
60u Frost Oil injected, temp doesn't go below -22C
60u Condensed Capscaicin injected, no temp change. Ingested, no temp change.

Reagent-wise, it's fine.

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #define HUMAN_MAX_OXYLOSS 1 //Defines how much oxyloss humans can get per tick. A tile with no air at all (such as space) applies this value, otherwise it's a percentage of it.
 #define HUMAN_CRIT_MAX_OXYLOSS ( 2.0 / 6) //The amount of damage you'll get when in critical condition. We want this to be a 5 minute deal = 300s. There are 50HP to get through, so (1/6)*last_tick_duration per second. Breaths however only happen every 4 ticks. last_tick_duration = ~2.0 on average
 
-#define HEAT_DAMAGE_LEVEL_1 2 //Amount of damage applied when your body temperature just passes the 360.15k safety point
+#define HEAT_DAMAGE_LEVEL_1 5 //Amount of damage applied when your body temperature just passes the 360.15k safety point
```
Comment (Anewbe) [inline code/modules/mob/living/carbon/human/life.dm]: It was at 5 at one point in the past and there were issues, people were displeased.

File: code/modules/mob/living/carbon/human/life.dm [hunk 2]
```diff
@@ -1181,7 +1185,7 @@
 
 				// Apply a fire overlay if we're burning.
 				if(on_fire)
-					health_images += image('icons/mob/screen1_health.dmi',"burning")
+					health_images += image('icons/mob/OnFire.dmi',"[get_fire_icon_state()]")
```
Comment (Anewbe) [inline code/modules/mob/living/carbon/human/life.dm]: Is this the massive /tg/ fire thing? Still not sure how I feel about that.

File: code/modules/mob/living/living_defense.dm [hunk 3]
```diff
@@ -351,6 +351,24 @@
...
+		fire_stacks /= 2
+		L.fire_stacks += fire_stacks
+		if(L.IgniteMob())
+			message_admins("[key_name(src)] bumped into [key_name(L)] and set them on fire.")
+
+	if(L_old_on_fire) //Only ignite us and gain their stacks if they were onfire before we bumped them
+		L.fire_stacks /= 2
+		fire_stacks += L.fire_stacks
+		IgniteMob()
+
```
Comment (Anewbe) [inline code/modules/mob/living/living_defense.dm]: This concept will probably need to be discussed.

**Comment by SilverTalismen** [comment]
Hi could I see the on fire sprite? :o If you don't mind?

**Comment by Nalarac** [comment]
This would Fix #3817 since that issue was about autoheal which is being changed here.

**Comment by Anewbe** [comment]
Neerti knows what needs to be done with this PR.

**Comment by Anewbe** [comment]
Oh boy, here we go.

**Pull request merged**